README.md

```diff
@@ -21,12 +21,20 @@ npm install fsevents
 
 ```js
 const fsevents = require('fsevents');
+
+// To start observation
 const stop = fsevents.watch(__dirname, (path, flags, id) => {
   const info = fsevents.getInfo(path, flags, id);
-}); // To start observation
-stop(); // To end observation
+});
+
+// To end observation
+stop();
 ```
 
+> **Important note:** The API behaviour is slightly different from typical JS APIs. The `stop` function **must** be
+> retrieved and stored somewhere, even if you don't plan to stop the watcher. If you forget it, the garbage collector
+> will eventually kick in, the watcher will be unregistered, and your callbacks won't be called anymore.
+
```

Comment:
I'm wondering, maybe this note could be reflected in the sample code?
Something like:

```js
const watchers = {};

const logAndWatch = (path) => {
    console.log('Watching ' + path)
    const stop = fsevents.watch(__dirname, (path, flags, id) => {
        const info = fsevents.getInfo(path, flags, id);
        console.dir(info);
    });
    watchers[path] = stop;
}

// Start observation and store the watcher
let myPath = '/path/to/somefile.txt';
logAndWatch(myPath);

// End observation, and get rid of the watcher
watchers[myPath]();
watchers[myPath] = null;
```

It's not as sweet as the original 4-liner, but can prevent some head scratching.

It seems inevitable to have to use an array or object to store the stop reference, otherwise in a local function the ref will go out of scope.